app/src/ui/lib/popover.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react'
+import FocusTrap from 'focus-trap-react'
+import { Options as FocusTrapOptions } from 'focus-trap'
+
+/**
+ * Position of the caret relative to the pop up. It's composed by 2 dimensions:
+ * - The first one is the edge on which the caret will rest.
+ * - The second one is the alignment of the caret within that edge.
+ *
+ * Example: TopRight means the caret will be in the top edge, on its right side.
+ **/
+export enum PopoverCaretPosition {
+  TopRight = 'top-right',
+  LeftTop = 'left-top',
+}
+interface IPopoverProps {
+  readonly onClickOutside: () => void
+  readonly caretPosition: PopoverCaretPosition
+}
+
+export class Popover extends React.Component<IPopoverProps> {
+  private focusTrapOptions: FocusTrapOptions
+  private divRef = React.createRef<HTMLDivElement>()
+
+  public constructor(props: IPopoverProps) {
+    super(props)
+
+    this.focusTrapOptions = {
+      allowOutsideClick: true,
+      escapeDeactivates: true,
+      onDeactivate: this.props.onClickOutside,
+    }
```

Comment:
Popover being a more generic component. What do you think about having focusTrapOptions be an optional input? so later we could have the flexibility to specify one of the other options like 'onActivate'?

Reply:
That sounds good, although I usually prefer not adding stuff until we need it (unless we know we're gonna need it 100%). An example of that are the caret positions: I only added the two we know we need (one for the diff options, already in use, and the other one for the new popover I'm working on). We keep the API simpler and we prevent adding complexity before we need it.

In the future, when (if) we need it, we can add that option.

At least that's how I see it 😅  Does that sound ok to you?